src/hotspot/cpu/ppc/frame_ppc.inline.hpp

```diff
@@ -33,10 +33,18 @@
 // Inline functions for ppc64 frames:
 
 // Find codeblob and set deopt_state.
-inline void frame::find_codeblob_and_set_pc_and_deopt_state(address pc) {
+inline void frame::find_codeblob_and_set_pc_and_deopt_state(address pc, bool forSignalHandler) {
   assert(pc != NULL, "precondition: must have PC");
 
-  _cb = CodeCache::find_blob(pc);
+  if (!forSignalHandler) {
```

Comment:
To reduce the duplication I suggest adding a helper method to CodeCache:
```
CodeBlob* find_blob(address pc, bool for_signal_handler) {
  if (!for_signal_handler) {
    return find_blob(pc);
  } else {
    CodeBlob* tmp = find_blob_unsafe(pc);
    if (tmp != NULL && temp->is_zombie()) {
      tmp = null;
    }
    return tmp;
}
```

Reply:
Done